lib/TokenGenerator/RandomTokenGenerator.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace OAuth2\TokenGenerator;
+
+use OAuth2\Model\IOAuth2Client;
+
+class RandomTokenGenerator implements TokenGeneratorInterface
+{
+    public function genAccessToken(IOAuth2Client $client, $data, $scope = null, $access_token_lifetime = null, $issue_refresh_token = true, $refresh_token_lifetime = null)
```

Comment:
This parameters should be extraced as class members and set over an constructor or setter methods.